src/main/scala/devices/debug/Debug.scala

```diff
@@ -644,16 +646,23 @@ class TLDebugModuleOuterAsync(device: Device)(implicit p: Parameters) extends La
     })
     val rf_reset = IO(Input(Reset()))
 
-    dmi2tlOpt.foreach { _.module.io.dmi <> io.dmi.get }
+    childClock := io.dmi_clock
+    childReset := io.dmi_reset
 
-    dmiBypass.module.io.bypass := ~io.ctrl.dmactive | ~AsyncResetSynchronizerShiftReg(in=io.ctrl.dmactiveAck, sync=3, name=Some("dmactiveAckSync"))
+    withClockAndReset(childClock, childReset) {
+      dmOuter.module.clock := io.dmi_clock
+      dmOuter.module.reset := io.dmi_reset
```

Comment:
```suggestion
```

I *think* that since you are basically assigning these to `childClock/childReset`, you don't need to say that. That is the rationale for having `childClock` and `childReset`, all Lazy submodules should get that clock/reset automatically. Did you try without this and have any issues?

Reply:
You are correct - fixed.